h/views/admin/search.py

```diff
@@ -25,9 +37,50 @@ def reindex_date(self):
         self.request.find_service(name="search_index").add_annotations_between_times(
             start_time, end_time, tag="reindex_date"
         )
+        return self._notify_reindexing_started(
+            f"Began reindexing from {start_time} to {end_time}"
+        )
 
-        self.request.session.flash(
-            f"Began reindexing from {start_time} to {end_time}", "success"
+    @view_config(
+        request_method="POST",
+        request_param="reindex_user",
+        require_csrf=True,
+        renderer="h:templates/admin/search.html.jinja2",
+    )
+    def reindex_user(self):
+        username = self.request.params["username"].strip()
+        user = models.User.get_by_username(
+            self.request.db, username, self.request.default_authority
+        )
```

Comment:
There is currently no way to specify a user at a non-default authority. I think that's out of scope for this PR.